src/stk500.c

```diff
@@ -1402,20 +1450,19 @@ static void stk500_print_parms1(const PROGRAMMER *pgm, const char *p, FILE *fp)
       }
       f /= prescale;
       f /= (osc_cmatch + 1);
-      if (f > 1e6) {
-        f /= 1e6;
-        unit = "MHz";
-      } else if (f > 1e3) {
-        f /= 1000;
-        unit = "kHz";
-      } else
-        unit = "Hz";
-      fmsg_out(fp, "%.3f %s\n", f, unit);
+      decimals = get_decimals(f);
+      f = f_to_kHz_MHz(f, &unit);
+      fmsg_out(fp, "%.*f %s\n", f, decimals, unit);
```

Comment:
Must be `fmsg_out(fp, "%.*f %s\n", decimals, f, unit);` ie, decimals first.

Reply:
Right, star comes before f in the string - I know I checked it later with the C standard, but only corrected line 1465 below and forgot the other occurrences here and in v2.
Funny thing is that gcc doesn't complain when I do it the wrong way round and does *"what I mean"* - I don't know what clang does though.

Reply:
> does *"what I mean"*

A fluke maybe? Maybe doubles are passed in a different frame to functions than smaller data types via the stack?

Reply:
> gcc doesn't complain

gcc does not realise that the called function resolves to or calls printf()